chore: adds root module example + more docs (#1)
masterpointio/infra-monorepo-template: root-module/README.md

# root-module

> This is a dummy root module.

Reviewer: ```suggestion
This is a template root module.
```

> ## Recommenations (DO NOT INCLUDE THIS INTO THE REAL README)

Reviewer: ```suggestion
## Documentation Recommendations (DO NOT INCLUDE THIS INTO THE REAL README)
```

> - Nodule description: Briefly explain what the root module sets up (e.g., infrastructure for a production environment).

Reviewer: ```suggestion
- Module description: Briefly explain what the root module sets up (e.g., infrastructure for RDS Postgres instances).
```

> - Usage instructions: Include examples of how to run terraform init, terraform plan, and terraform apply.

Reviewer: I think we cut this. This documentation should be higher up in the team's docs, not in each root module's README
```suggestion
```

- Documentation of variables and outputs: generate variables (from variables.tf) and outputs (from outputs.tf).

> - Link to child modules and external resources: If the root module references child modules, mention where they’re located and link to their documentation.

Reviewer: Let's combine this since this is what TF-docs covers. 
```suggestion
- Use terraform-docs to ensure that variables, outputs, child module, and resource documentation is included. 
```
